util/types/time_test.go

```diff
@@ -71,6 +71,10 @@ func (s *testTimeSuite) TestDateTime(c *C) {
 		c.Assert(t.String(), Equals, test.Expect)
 	}
 
+	t, _ := ParseTime("121231113045.9999999", mysql.TypeDatetime, 6)
```

Comment:
No, the check fails to detect the error because time.String() has bug,
So I compare `t.Time.Second()` and `t.Time.Microsecond()` here directly.